rasp/base.py

```diff
@@ -85,6 +88,8 @@ def get_page_source(self, url, params=None, headers=None):
         """
         if not url:
             raise ValueError('url needs to be specified')
+        if pre_fetch_callback:
+            pre_fetch_callback()
```

Comment:
Do we want this to have a say in the control flow (i.e. return a boolean used to determine whether or not to send the get request)?